### PR TITLE
Add a PARSING documentation

### PR DESCRIPTION
**Pull request by mwestphal**
- Add a PARSING.md doc in the user doc
- Add links to it all over the doc
- Fix a few doc issues
- Fix https://github.com/f3d-app/f3d/issues/1572

### CODE REVIEW

File: doc/libf3d/OPTIONS.md [hunk 1]
```diff
@@ -43,9 +43,9 @@ Option|Type<br>Default<br>Trigger|Description|F3D option
 :---:|:---:|:---|:---:
 model.matcap.texture|string<br>optional<br>render|Path to a texture file containing a material capture. All other model options for surfaces are ignored if this is set. Model specified by default.|\-\-texture-matcap
 model.color.opacity|double<br>optional<br>render|Set *opacity* on the geometry. Usually used with Depth Peeling option. Multiplied with the `model.color.texture` when present. Model specified by default.|\-\-opacity
-model.color.rgb|vector\<double\><br>optional<br>render|Set a *color* on the geometry. Multiplied with the `model.color.texture` when present. Model specified by default.|\-\-color
+model.color.rgb|double_vector<br>optional<br>render|Set a *color* on the geometry. Multiplied with the `model.color.texture` when present. Model specified by default.|\-\-color
```
Comment (mwestphal) [inline doc/libf3d/OPTIONS.md:46]: Im not sure this is the right syntax, wdyt @Meakk ?

Comment (Meakk) [reply doc/libf3d/OPTIONS.md:46]: Wdym?

File: doc/user/PARSING.md [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+ - yes/no
+ - on/off
+ - 1/0
+
+When formatting a bool into a string, true/false is used.
+
+## Int
+
+Int parsing is supported using [std::stoi](https://en.cppreference.com/w/cpp/string/basic_string/stol) and check
+that the whole string is parsed.
```
Comment (Meakk) [inline doc/user/PARSING.md:27]: What happens if the check fails? Document

File: doc/libf3d/OPTIONS.md (hunk 1 above)
Comment (mwestphal) [reply doc/libf3d/OPTIONS.md:46]: Should we write `double_vector` or `vector<double>` in the user doc ?

File: doc/user/PARSING.md (hunk 2 above)
Comment (mwestphal) [reply doc/user/PARSING.md:27]: the value is not changed as parsing fails, I should document this behavior more generally though

Comment (mwestphal) [reply doc/user/PARSING.md:27]: I\ve added a note about parsing failure but I cant really be more precise, as the exact behavior depends if the user uses the libf3d or F3D

File: doc/libf3d/OPTIONS.md (hunk 1 above)
Comment (Meakk) [reply doc/libf3d/OPTIONS.md:46]: `double[3]`?

File: doc/user/PARSING.md [hunk 3]
```diff
@@ -0,0 +1,58 @@
+# Parsing options
+
+When setting options from the [CLI Options](OPTIONS.md), the [commands](COMMANDS.md) or using the [libf3d options string API](../libf3d/OPTIONS.md#string-api), the values are parsed according to their type. If parsing fails, the values is not changed.
```
Comment (Meakk) [inline doc/user/PARSING.md:3]:
```suggestion
When setting options from the [CLI Options](OPTIONS.md), the [commands](COMMANDS.md) or using the [libf3d options string API](../libf3d/OPTIONS.md#string-api), the values are parsed according to their type. If parsing fails, the value is not changed.
```

File: doc/libf3d/OPTIONS.md (hunk 1 above)
Comment (mwestphal) [reply doc/libf3d/OPTIONS.md:46]: nope, that would be incorrect, it can receive any number of values.

Comment (mwestphal) [reply doc/libf3d/OPTIONS.md:46]: also it even more programmation oriented, which our f3d users are not

File: doc/user/PARSING.md [hunk 4]
```diff
@@ -0,0 +1,59 @@
+# Parsing options
+
+When setting options from the [CLI Options](OPTIONS.md), the [commands](COMMANDS.md) or using the [libf3d options string API](../libf3d/OPTIONS.md#string-api), the values are parsed according to their type. If parsing fails, the value is not changed.
+
+The following types are supported:
+ - bool: A boolean, true or false.
+ - int: An signed integer.
```
Comment (Meakk) [inline doc/user/PARSING.md:7]:
```suggestion
 - int: A signed integer.
```

**Review by Meakk — APPROVED** [review]

**Pull request merged**